### PR TITLE
Fix gitlab.com detection due to CloudFlare

### PR DESCRIPTION
**Pull request by Kidswiss**
The check to determine if it's a gitlab instance or not will query the
`/users/sign_in` path. If that path is not found it's not a GitLab
instance. Gitlab.com is using some CloudFlare browser detection on that
path, which works very well and blocks that request from the operator.

Now the operator will determine that it's a GitLab instance if the
endpoint contains `gitlab.com`.

Closes: #83

### CODE REVIEW

File: pkg/git/gitlab/gitlab.go [hunk 1]
```diff
@@ -230,23 +230,27 @@ func (g *Gitlab) FullURL() *url.URL {
...
 		Timeout: 10 * time.Second,
 	}
 
-	gitlabURL := URL.Scheme + "://" + URL.Host + "/users/sign_in'"
+	gitlabURL := URL.Scheme + "://" + URL.Host + "/users/sign_in"
 
 	resp, err := httpClient.Get(gitlabURL)
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("can't query gitlab login page: %v", err)
```
Comment (srueg) [inline pkg/git/gitlab/gitlab.go:245]:
```suggestion
		return false, fmt.Errorf("can't query gitlab login page: %w", err)
```

File: pkg/git/gitlab/gitlab.go [hunk 2]
```diff
@@ -230,23 +230,27 @@ func (g *Gitlab) FullURL() *url.URL {
 // it will return false.
 func (g *Gitlab) IsType(URL *url.URL) (bool, error) {
 
+	if strings.Contains(URL.Host, "gitlab.com") {
```
Comment (srueg) [inline pkg/git/gitlab/gitlab.go:233]: I really don't like using such "magic strings".
We could use the [/version](https://docs.gitlab.com/ee/api/version.html) endpoint?

Comment (Kidswiss) [reply pkg/git/gitlab/gitlab.go:233]: It's not available on gitlab.com:
```
➜ curl https://gitlab.com/version
<html><body>You are being <a href="https://gitlab.com/users/sign_in">redirected</a>.</body></html>%    
```

File: pkg/git/gitlab/gitlab.go (hunk 1 above)
Comment (srueg) [reply pkg/git/gitlab/gitlab.go:245]: What's the reason to stick with `%v` here?

File: pkg/git/gitlab/gitlab.go (hunk 2 above)
Comment (srueg) [reply pkg/git/gitlab/gitlab.go:233]: Another idea would be to just return `true` and think about this problem again once we actually implement other Git providers.

I'm not even sure anymore it makes sense to auto discover the git provider like this. We might as well just include another field in the credentials secret to state which git provider it is. This just seems very brittle and probably causes more errors and confusion than it helps.

What do you think?

Comment (Kidswiss) [reply pkg/git/gitlab/gitlab.go:233]: Fine by me.

Until we implement the change correclty I'll do the `return true` thing.

File: pkg/git/gitlab/gitlab.go (hunk 1 above)
Comment (Kidswiss) [reply pkg/git/gitlab/gitlab.go]: Habit. Changed it to `%w`.

**Comment by srueg** [comment]
Now the test `TestGitlab_IsType/is_not_Gitlab` fails 🙈

**Comment by Kidswiss** [comment]
I completely removed the `TestGitlab_IsType` now as the function doesn't really do anything anymore.

We should open a new issue to outline how we want to handle the git type definition later on. I don't think that has much prio now as we're only supporting GitLab anyway.

**Pull request merged**